api_test.go

```diff
@@ -1749,3 +1751,126 @@ func TestCreateSiteRulesRateLimitClientIdentifiers(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestCRUDSiteRequestRule(t *testing.T) {
+	createSiteRulesBody := CreateSiteRuleBody{
+		Type:           "request",
+		GroupOperator:  "all",
+		Enabled:        true,
+		Reason:         "Example site rule",
+		RequestLogging: "none",
+		Expiration:     "",
+		Conditions: []Condition{
+			{
+				Type:     "single",
+				Field:    "ip",
+				Operator: "equals",
+				Value:    "1.2.3.4",
+			},
+			{
+				Type:          "group",
+				GroupOperator: "any",
+				Conditions: []Condition{
+					{
+						Type:     "single",
+						Field:    "ip",
+						Operator: "equals",
+						Value:    "5.6.7.8",
+					},
+				},
+			},
+		},
+		Actions: []Action{
+			{
+				Type: "block",
+			},
+		},
+	}
+
+	sc := NewTokenClient(testcreds.email, testcreds.token)
+	corp := testcreds.corp
+	site := testcreds.site
+
+	createResp, err := sc.CreateSiteRule(corp, site, createSiteRulesBody)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !compareSiteRuleBody(createSiteRulesBody, createResp.CreateSiteRuleBody) {
+		t.Errorf("CreateSiteRules got:\n %#v\n want\n %#v", createResp, createSiteRulesBody)
+	}
+
+	readResp, err := sc.GetSiteRuleByID(corp, site, createResp.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !compareSiteRuleBody(createSiteRulesBody, readResp.CreateSiteRuleBody) {
+		t.Errorf("CreateSiteRules got:\n %#v\n want\n %#v", createResp, createSiteRulesBody)
+	}
+
+	updateSiteRuleBody := CreateSiteRuleBody{
+		Type:           "request",
+		GroupOperator:  "all",
+		Enabled:        true,
+		Reason:         "Example site rule",
+		Expiration:     "",
+		RequestLogging: "none",
+		Conditions: []Condition{
+			{
+				Type:     "single",
+				Field:    "ip",
+				Operator: "equals",
+				Value:    "1.2.3.4",
+			},
+			{
+				Type:          "group",
+				GroupOperator: "any",
+				Conditions: []Condition{
+					{
+						Type:     "single",
+						Field:    "ip",
+						Operator: "equals",
+						Value:    "9.10.11.12",
+					},
+				},
+			},
+		},
+		Actions: []Action{
+			{
+				Type: "block",
+			},
+		},
+	}
+
+	updateResp, err := sc.UpdateSiteRuleByID(corp, site, createResp.ID, updateSiteRuleBody)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !compareSiteRuleBody(updateSiteRuleBody, updateResp.CreateSiteRuleBody) {
+		t.Errorf("CreateSiteRules got:\n %#v\n want %#v", updateResp, updateSiteRuleBody)
+	}
+
+	readall, err := sc.GetAllSiteRules(corp, site)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(readall.Data) != 1 {
+		t.Error()
+	}
+
+	if readall.TotalCount != 1 {
+		t.Error()
+	}
```

Comment:
There's a case where a site might have a rule of `type:templatedSignal`. These rules don't show in the rules UI.  You see them in the newer templated rules part of the UI. If at least one of those rules are configured on the site then this test case is always going to be wrong - I know only because I have one set up on my personal account and spent way too much time this morning trying to figure out these tests wouldn't pass.

So I suggest just making sure that the `Data` and `TotalCount` lengths are only non-zero.  I might even go as far as suggesting to remove these length assertions only because the point of these tests isn't to exercise the SigSci api, but to make sure this client sdk is hooked up properly.

```suggestion
	if len(readall.Data) != 0 {
		t.Error()
	}

	if readall.TotalCount != 0 {
		t.Error()
	}
```